### PR TITLE
Changes for llvm3.9

### PR DESCRIPTION
**Pull request by sklam**
Depends on https://github.com/numba/llvmlite/pull/233.

### CODE REVIEW

File: numba/targets/codegen.py [hunk 1]
```diff
@@ -531,16 +531,13 @@ def unserialize_library(self, serialized):
 
     def _module_pass_manager(self):
         pm = ll.create_module_pass_manager()
-        dl = ll.create_target_data(self._data_layout)
-        dl.add_pass(pm)
```
Comment (sklam) [inline numba/targets/codegen.py]: `TargetData.add_pass` and its C++ impl are removed in LLVM3.9.

**Comment by seibert** [comment]
This patch is pretty tiny.  If CI passes, I think this is ready to merge.  We have already bumped our llvmlite requirement to >= 0.16, right?

**Comment by sklam** [comment]
TravisCI passed the build with the latest llvmlite using llvm 3.9.1

File: numba/cuda/cudadrv/nvvm.py [hunk 2]
```diff
@@ -496,12 +498,15 @@ def parse_out_leading_type(s):
...
     buf = []
     for line in ir.splitlines():
-        orig = line
         if re_metadata_def.match(line):
             # Rewrite metadata since LLVM 3.7 dropped the "metadata" type prefix.
             if None is re_metadata_correct_usage.search(line):
                 line = line.replace('!{', 'metadata !{')
                 line = line.replace('!"', 'metadata !"')
+        if line.startswith('source_filename ='):
+            line = ''   # skip line
```
Comment (stuartarchibald) [inline numba/cuda/cudadrv/nvvm.py]: could this just be `continue` to jump over the other conditions?

**Pull request merged**